File: lib/point.js
/**
 * Declaration of Point(x, y), one of the basic structures needed
 */
var Point = function(x, y) {
    this.x = x; // coordinate by x-axis
    this.y = y; // coordinate by y-axis
};

/**
 * Shift Point on Vector
 */
var shiftPoint = function(p, v) {
    // we should add x and y coordiantes of vector
    // respectively to x and y coordiantes of point
    return new Point(p.x + v.x, p.y + v.y);
};

File: slines_py/lib/point.py
'''
Declaration of Point(x, y), one of the basic structures needed
'''
class Point:
    def __init__(self, _x = 0, _y = 0):
        self.x = _x
        self.y = _y

    def __add__(self, _vector):
        return Point(self.x + _vector.x, self.y + _vector.y)

    def __iadd__(self, _vector):
        self.x += _vector.x
        self.y += _vector.y


'''
Declaration of fakePoint, used instead of Point in the holes
'''
class FakePoint(Point):
    def __init__(self):
        Point.__init__(self, 0, 0)